tests/integration/suite/daprd/workflow/reconnect.go

```diff
@@ -0,0 +1,197 @@
+				return nil, err
+			}
+			return fmt.Sprintf("Hello, %s!", name), nil
+		})
+		taskhubCtx, cancelTaskhub := context.WithCancel(ctx)
+		require.NoError(t, backendClient.StartWorkItemListener(taskhubCtx, r))
+		defer cancelTaskhub()
+
+		id, err := backendClient.ScheduleNewOrchestration(ctx, "ReconnectDuringWaitForEvent", api.WithInstanceID("Dapr"), api.WithInput("Dapr"))
+		require.NoError(t, err)
+
+		_, err = backendClient.WaitForOrchestrationStart(ctx, id)
+		require.NoError(t, err)
+
+		cancelTaskhub() // disconnect the listener
+
+		taskhubCtx, cancelTaskhub = context.WithCancel(ctx)
+		require.NoError(t, backendClient.StartWorkItemListener(taskhubCtx, r))
+		defer cancelTaskhub()
+
+		_, err = backendClient.WaitForOrchestrationStart(ctx, id)
+		require.NoError(t, err)
+
+		err = backendClient.RaiseEvent(ctx, id, "event1")
+		require.NoError(t, err)
+
+		metadata, err := backendClient.WaitForOrchestrationCompletion(ctx, id, api.WithFetchPayloads(true))
+		require.NoError(t, err)
+		assert.Equal(t, api.RUNTIME_STATUS_COMPLETED, metadata.RuntimeStatus)
+		assert.Equal(t, `"Hello, Dapr!"`, metadata.SerializedOutput)
+	})
+
+	t.Run("reconnect_during_activity", func(t *testing.T) {
+		numActivities := 5
+		activities := make(chan struct{}, numActivities)
+		for range numActivities - 1 {
+			activities <- struct{}{}
+		}
+
+		r := task.NewTaskRegistry()
+		r.AddOrchestratorN("ReconnectDuringActivity", func(ctx *task.OrchestrationContext) (any, error) {
+			var input string
+			if err := ctx.GetInput(&input); err != nil {
+				return nil, err
+			}
+			var output string
+			for range numActivities {
+				err := ctx.CallActivity("SayHello", task.WithActivityInput(input)).Await(&output)
+				if err != nil {
+					return nil, err
+				}
+			}
+
+			return output, nil
+		})
+		r.AddActivityN("SayHello", func(ctx task.ActivityContext) (any, error) {
+			var name string
+			if err := ctx.GetInput(&name); err != nil {
+				return nil, err
+			}
+
+			<-activities // the last activity will block until after the reconnection
+
+			return fmt.Sprintf("Hello, %s!", name), nil
+		})
+		taskhubCtx, cancelTaskhub := context.WithCancel(ctx)
+		require.NoError(t, backendClient.StartWorkItemListener(taskhubCtx, r))
+		defer cancelTaskhub()
+
+		id, err := backendClient.ScheduleNewOrchestration(ctx, "ReconnectDuringActivity", api.WithInstanceID("Dapr"), api.WithInput("Dapr"))
+		require.NoError(t, err)
+
+		_, err = backendClient.WaitForOrchestrationStart(ctx, id)
+		require.NoError(t, err)
+
+		cancelTaskhub() // disconnect the listener
+
+		taskhubCtx, cancelTaskhub = context.WithCancel(ctx)
+		require.NoError(t, backendClient.StartWorkItemListener(taskhubCtx, r))
+		defer cancelTaskhub()
+
+		_, err = backendClient.WaitForOrchestrationStart(ctx, id)
+		require.NoError(t, err)
+
+		activities <- struct{}{} // release the last activity
+
+		metadata, err := backendClient.WaitForOrchestrationCompletion(ctx, id, api.WithFetchPayloads(true))
```

Comment:
Should this get wrapped in an assert eventually? I see it timing out in CI due to `context deadline exceeded` on 
 Windows.

For Mac I also see `Workflow actor 'Dapr': workflow was previously completed and is being recreated` && `Dapr: aborting workflow execution` where that is apparently retriable, but its not shutting down placement properly within the 45s granted for IT tests.

Reply:
is this something retried in the Dapr SDK (I'll check)? Ideally these tests would use the Dapr SDK but the existing ones didn't so I kept consistency.